### PR TITLE
LPD-23754 validation added to avoid updating account entries with empty values

### PR DESCRIPTION
**Pull request by brbalazs**
https://liferay.atlassian.net/browse/LPD-23754

### CODE REVIEW

File: modules/apps/commerce/headless/headless-commerce/headless-commerce-admin-account-impl/src/main/java/com/liferay/headless/commerce/admin/account/internal/resource/v1_0/AccountResourceImpl.java [hunk 1]
```diff
@@ -486,22 +486,26 @@ private AccountEntry _updateAccount(Long id, Account account)
 			GetterUtil.getLong(
 				account.getDefaultBillingAccountAddressId(),
 				accountEntry.getDefaultBillingAddressId()));
+
```
Comment (smottal) [inline modules/apps/commerce/headless/headless-commerce/headless-commerce-admin-account-impl/src/main/java/com/liferay/headless/commerce/admin/account/internal/resource/v1_0/AccountResourceImpl.java:489]: please remove the line

File: modules/apps/commerce/headless/headless-commerce/headless-commerce-admin-account-impl/src/main/java/com/liferay/headless/commerce/admin/account/internal/resource/v1_0/AccountResourceImpl.java [hunk 2]
```diff
@@ -486,22 +486,26 @@ private AccountEntry _updateAccount(Long id, Account account)
...
 		accountEntry.setDefaultShippingAddressId(
 			GetterUtil.getLong(
 				account.getDefaultShippingAccountAddressId(),
 				accountEntry.getDefaultShippingAddressId()));
+
```
Comment (smottal) [inline modules/apps/commerce/headless/headless-commerce/headless-commerce-admin-account-impl/src/main/java/com/liferay/headless/commerce/admin/account/internal/resource/v1_0/AccountResourceImpl.java:494]: please remove the line

File: modules/apps/commerce/headless/headless-commerce/headless-commerce-admin-account-impl/src/main/java/com/liferay/headless/commerce/admin/account/internal/resource/v1_0/AccountResourceImpl.java [hunk 3]
```diff
@@ -486,22 +486,26 @@ private AccountEntry _updateAccount(Long id, Account account)
...
 		accountEntry.setEmailAddress(_getEmailAddress(account, accountEntry));
-		accountEntry.setName(account.getName());
+
```
Comment (smottal) [inline modules/apps/commerce/headless/headless-commerce/headless-commerce-admin-account-impl/src/main/java/com/liferay/headless/commerce/admin/account/internal/resource/v1_0/AccountResourceImpl.java:496]: please remove the line

File: modules/apps/commerce/headless/headless-commerce/headless-commerce-admin-account-impl/src/main/java/com/liferay/headless/commerce/admin/account/internal/resource/v1_0/AccountResourceImpl.java [hunk 4]
```diff
@@ -486,22 +486,26 @@ private AccountEntry _updateAccount(Long id, Account account)
...
+		accountEntry.setName(
+			GetterUtil.get(account.getName(), accountEntry.getName()));
+
```
Comment (smottal) [inline modules/apps/commerce/headless/headless-commerce/headless-commerce-admin-account-impl/src/main/java/com/liferay/headless/commerce/admin/account/internal/resource/v1_0/AccountResourceImpl.java:499]: please remove the line

File: modules/apps/commerce/headless/headless-commerce/headless-commerce-admin-account-impl/src/main/java/com/liferay/headless/commerce/admin/account/internal/resource/v1_0/AccountResourceImpl.java [hunk 5]
```diff
@@ -486,22 +486,26 @@ private AccountEntry _updateAccount(Long id, Account account)
...
 		accountEntry.setTaxIdNumber(
 			GetterUtil.get(account.getTaxId(), accountEntry.getTaxIdNumber()));
+
```
Comment (smottal) [inline modules/apps/commerce/headless/headless-commerce/headless-commerce-admin-account-impl/src/main/java/com/liferay/headless/commerce/admin/account/internal/resource/v1_0/AccountResourceImpl.java:502]: please remove the line

**Review by smottal — CHANGES_REQUESTED** [review]

**Comment by brbalazs** [comment]
ci:test:sf

File: modules/apps/commerce/headless/headless-commerce/headless-commerce-admin-account-impl/src/main/java/com/liferay/headless/commerce/admin/account/internal/resource/v1_0/AccountResourceImpl.java [hunk 6]
```diff
@@ -500,8 +501,6 @@ private AccountEntry _updateAccount(Long id, Account account)
 
 		accountEntry = _accountEntryService.updateAccountEntry(accountEntry);
 
-		// Expando
-
```
Comment (smottal) [inline modules/apps/commerce/headless/headless-commerce/headless-commerce-admin-account-impl/src/main/java/com/liferay/headless/commerce/admin/account/internal/resource/v1_0/AccountResourceImpl.java:504]: please put again the two comments

**Review by smottal — CHANGES_REQUESTED** [review]

**Review by smottal — APPROVED** [review]

**Comment by smottal** [comment]
ci:test:relevant

**Comment by smottal** [comment]
ci:test:um

**Comment by smottal** [comment]
ci:report:2562471245

**Comment by smottal** [comment]
ci:forward

**Pull request closed without merging**